frontend/pages/api/search.ts

```diff
@@ -0,0 +1,21 @@
+import MeiliSearch from 'meilisearch';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { MemberHit } from '~/types/MemberHit';
+
+const client = new MeiliSearch({
+  host: process.env.MEILI_HOST,
+  apiKey: process.env.MEILI_MASTER_KEY,
+});
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== 'GET') {
+    res.status(404);
+  }
+
+  const searchQuery = req.query.q as string;
+  const escapedQuery = searchQuery.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+
+  const result = await client.index('members').search<MemberHit>(escapedQuery);
+
+  return res.status(200).json(result);
+}
```

Comment:
Are you sure this always happens server side? (I'm don't really know the rules for SSR/CSR in Next) Is there any way we can put some restraints ensuring you can't easily change the code to be run client side?

Reply:
It is always server side, since it is in the pages/api folder which makes the master key not public to the browser and still takes advantage of the better performance of using the meilisearch client directly instead of using a custom service and graphql.

Reply:
@emilwihlander the restraint that makes sure this cannot run client side is the fact that we do not use NEXT_PUBLIC environment variables.